### PR TITLE
Add calendar factory for the schedule_function API

### PR DESCRIPTION
**Pull request by dmichalowicz**

### CODE REVIEW

File: zipline/utils/events.py [hunk 1]
```diff
@@ -603,6 +605,24 @@ class time_rules(object):
...
+    @property
+    def us_equities(self):
+        return get_calendar('NYSE')
+
+    @property
+    def us_futures(self):
+        return get_calendar('us_futures')
+
+
+class calendars(six.with_metaclass(CalendarsMeta)):
```
Comment (ssanderson) [inline zipline/utils/events.py]: FYI, we also have `zipline.utils.classproperty`.

Comment (dmichalowicz) [inline zipline/utils/events.py]: Oh that's perfect! Thanks for pointing that out

File: zipline/algorithm.py [hunk 2]
```diff
@@ -1106,7 +1109,12 @@ def schedule_function(self,
         # Check the type of the algorithm's schedule before pulling calendar
         # Note that the ExchangeTradingSchedule is currently the only
         # TradingSchedule class, so this is unlikely to be hit
-        cal = calendar or self.trading_calendar
+        if calendar is calendars.US_EQUITIES:
+            cal = get_calendar('NYSE')
+        elif calendar is calendars.US_FUTURES:
+            cal = get_calendar('us_futures')
+        else:
```
Comment (richafrank) [inline zipline/algorithm.py]: Should this explicitly check for `calendar is None`? I would expect that passing some other value should raise an exception or warn me that it's being ignored.

File: zipline/api.pyi [hunk 3]
```diff
@@ -545,6 +545,8 @@ def schedule_function(func, date_rule=None, time_rule=None, half_days=True, cale
         The rule for the times to execute this function.
     half_days : bool, optional
         Should this rule fire on half days?
+    calendar : Sentinel, optional
```
Comment (richafrank) [inline zipline/api.pyi]: FYI We regenerate this file when we [release](http://www.zipline.io/release-process.html#updating-the-python-stub-files) a new version of zipline.

File: zipline/utils/events.py [hunk 4]
```diff
@@ -603,6 +605,11 @@ class time_rules(object):
     every_minute = Always
 
 
+class calendars(object):
+    US_EQUITIES = sentinel('us_equities_calendar')
```
Comment (richafrank) [inline zipline/utils/events.py]: Nice use of `sentinel`.

File: zipline/api.py [hunk 5]
```diff
@@ -53,5 +54,6 @@
     'execution',
     'math_utils',
     'slippage',
-    'time_rules'
+    'time_rules',
+    'calendars',
```
Comment (richafrank) [inline zipline/api.py]: I didn't recall discussing this addition. @ssanderson This align with your understanding?

Comment (dmichalowicz) [inline zipline/api.py]: I guess we never did discuss this piece of it. I had added it here initially for consistency with `date_rules` and `time_rules`, but I'm totally open to changing this to a more appropriate place.

File: zipline/api.pyi (hunk 3 above)
Comment (dmichalowicz) [inline zipline/api.pyi]: Ok, so this addition is unnecessary then?

File: zipline/algorithm.py (hunk 2 above)
Comment (dmichalowicz) [inline zipline/algorithm.py]: Ah good catch. I'll add a check for `None`, and the `else` case should probably raise an exception.

File: zipline/api.pyi (hunk 3 above)
Comment (richafrank) [inline zipline/api.pyi]: Yea. It certainly doesn't hurt anything, but want you to know it's not required.

File: zipline/api.py (hunk 5 above)
Comment (richafrank) [inline zipline/api.py]: The consistency with `date_rules` and `time_rules` makes sense to me.

File: zipline/utils/events.py (hunk 4 above)
Comment (ssanderson) [inline zipline/utils/events.py]: Most of our uses of `sentinel` right now use the same string for both the variable name and the repr.  Any reason not to do that here?

File: zipline/api.py (hunk 5 above)
Comment (ssanderson) [inline zipline/api.py]: I'm :+1: on namespacing this under `zipline.api.calendars`.

File: zipline/utils/events.py (hunk 4 above)
Comment (dmichalowicz) [inline zipline/utils/events.py]: Nope, only that I didn't know that was the convention 😄 . I do think "us_equities_calendar" is a little clearer, but am also fine with changing it to be the same as "US_EQUITIES" if that's more consistent.

**Comment by dmichalowicz** [comment]
@richafrank Updated from your feedback

**Comment by richafrank** [comment]
👍 from me!

**Pull request merged**